src/main/java/com/afs/tdd/MarsRover.java

```diff
@@ -1,4 +1,57 @@
 package com.afs.tdd;
 
+import javafx.geometry.Pos;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.IntStream;
+
 public class MarsRover {
+
+    static List<String> directions = Arrays.asList("N", "E", "S", "W");
+
+    public Position executeBatchCommand(Position position, String batchCommand) {
+
+        for (String command: batchCommand.split("")){
+            position = executeCommand(position, command);
+        }
+        return position;
+    }
+
+    public Position executeCommand(Position position, String command) {
+
+        if (command.equals("M")) {
+            return move(position);
+        } else if (command.equals("R")) {
+            return turnRight(position);
+        } else if (command.equals("L")) {
+            return turnLeft(position);
+        }
+        return position;
+    }
+
+    private Position turnLeft(Position position) {
+        position.setDirection(directions.get((directions.indexOf(position.getDirection()) + 3) % 4));
+        return position;
+    }
+
+    private Position turnRight(Position position) {
+        position.setDirection(directions.get((directions.indexOf(position.getDirection()) + 1) % 4));
+        return position;
+    }
+
+
+
+    private Position move(Position position) {
+        String direction = position.getDirection();
+        if (direction == "N" || direction == "S") {
```

Comment:
seems a bit strange to merge these statements, probably can use switch to make it more clear